query.py

```diff
@@ -0,0 +1,153 @@
+import os
+import argparse
+import shelve
+from math import log2
+from gensim.parsing.porter import PorterStemmer
+from merge_operations import *

```

Comment:
Never (and never and never) use such imports, please, because of namespace overwriting. You could easily overwrite some functions and for hours try to understand what's going on.